nemo/collections/asr/parts/mixins/mixins.py

```diff
@@ -45,65 +45,153 @@ def _setup_tokenizer(self, tokenizer_cfg: DictConfig):
             os.environ['TOKENIZERS_PARALLELISM'] = 'false'
 
         self.tokenizer_cfg = OmegaConf.to_container(tokenizer_cfg, resolve=True)  # type: dict
-        self.tokenizer_dir = self.tokenizer_cfg.pop('dir')  # Remove tokenizer directory
-        self.tokenizer_type = self.tokenizer_cfg.pop('type').lower()  # Remove tokenizer_type
 
-        self.hf_tokenizer_kwargs = self.tokenizer_cfg.pop("hf_kwargs", {})  # Remove HF tokenizer kwargs
+        # the aggregate tokenizer does not have one tokenizer_dir but multiple ones
```

Comment:
The two codepaths here are complex enough to warrant splitting them into their own functions and calling them from this one.

Reply:
agreed and done